addons/udes_stock/tests/test_picking_batch.py

```diff
@@ -1032,6 +1032,157 @@ def test31_backorder_generated_unpickable_item_product_ok_in_two_picks_and_two_p
         self.assertEqual(len(inv_picking.move_line_ids), 1)
         self.assertEqual(inv_picking.move_line_ids.product_qty, 4)
 
+    def _assert_apple_banana_pickings(
+        self, pickings, num_pickings, state, move_list_qty, package_list=None
+    ):
+        """
+        Helper to avoid repeated asserts
+        Check the number of pickings, state and move list qty of the pickings provided
+        If the state is assigned also check the move lines
+        If the package_list is provided and the picking is assigned, check the packages
+        This is based on the symmetry of the test, that the pickings are
+        mirrored and differ only by product and packages
+        """
+        self.assertEqual(len(pickings), num_pickings)
+        self.assertEqual(pickings.mapped("state"), num_pickings * [state])
+        all_mvs = pickings.mapped("move_lines")
+        apple_mvs = all_mvs.filtered(lambda mv: mv.product_id == self.apple)
+        banana_mvs = all_mvs.filtered(lambda mv: mv.product_id == self.banana)
+        for mvs in [apple_mvs, banana_mvs]:
+            with self.subTest(mvs=mvs):
+                self.assertCountEqual(mvs.mapped("product_qty"), move_list_qty)
+                if state == "assigned":
+                    # If the state is assigned then also check the move lines
+                    mls = mvs.mapped("move_line_ids")
+                    self.assertCountEqual(mls.mapped("product_uom_qty"), move_list_qty)
+        if state == "assigned" and package_list:
+            all_mls = all_mvs.mapped("move_line_ids")
+            self.assertCountEqual([ml.package_id for ml in all_mls], package_list)
+
+    def test32_correctly_grouped_pickings_with_all_unpickable_items(self):
+        """
+        Tests that when flagging items in a batch as unpickable, stock investigations
+        are correctly raised with the backordered pickings correctly grouped,
+        if all the items are unpickable then we return to the original state of the system
+        """
+        Picking = self.env["stock.picking"]
+        Batch = self.env["stock.picking.batch"]
+        Group = self.env["procurement.group"]
+        # Cron job that deletes empty draft pickings
+        unlink_cron = self.env.ref("udes_stock.unlink_empty_pickings_action")
+        unlink_cron.active = False
+
+        # Create a list of packages used for the quants, and create a picking for each product
+        # that takes everything created
```

Comment:
Extend the comment to clarify that the packages contain 1, 2, 3 and 4 units of each product, for a total of 10.